framework/mesh/sweep_utilities/communicators/aah_asyn_comm.cc

```diff
@@ -280,34 +274,9 @@ AAH_ASynchronousCommunicator::ReceiveDelayedData(int angle_set_num)
 
         u_ll_int block_addr = delayed_prelocI_message_blockpos[prelocI][m];
         u_ll_int message_size = delayed_prelocI_message_size[prelocI][m];
-
-        int error_code = MPI_Recv(&upstream_psi[block_addr],
-                                  static_cast<int>(message_size),
-                                  MPI_DOUBLE,
-                                  comm_set_.MapIonJ(locJ, opensn::mpi.location_id),
-                                  max_num_mess * angle_set_num + m,
-                                  comm_set_.LocICommunicator(opensn::mpi.location_id),
-                                  MPI_STATUS_IGNORE);
+        comm.recv(source_rank, tag, &upstream_psi[block_addr], message_size);
 
         delayed_prelocI_message_received[prelocI][m] = true;
-
-        if (error_code != MPI_SUCCESS)
-        {
-          std::stringstream err_stream;
-          err_stream << "################# Delayed receive error."
-                     << " message size=" << message_size << " as_num=" << angle_set_num
-                     << " num_mess=" << num_mess << " m=" << m << " error="
-                     << " size="
-                     << "\n";
-          char error_string[BUFSIZ];
-          int length_of_error_string, error_class;
-          MPI_Error_class(error_code, &error_class);
-          MPI_Error_string(error_class, error_string, &length_of_error_string);
-          err_stream << error_string << "\n";
-          MPI_Error_string(error_code, error_string, &length_of_error_string);
-          err_stream << error_string << "\n";
-          log.LogAllWarning() << err_stream.str();
-        }
```

Comment:
how is the error handled now?

Reply:
This was an interesting observation. The default MPI behavior is that if an MPI call fails, MPI exits with an error and prints the error message, so you never see an error code other then MPI_SUCCESS unless you explicitly change the MPI behavior via [MPI_Comm_set_errhandler](https://www.mpich.org/static/docs/v1.5.x/www3/MPI_Comm_set_errhandler.html).

However, what the MPI wrapper does is that every MPI call is wrapped in a `check_mpi_error` call which check for an error and terminates the run, so application code does not have to worry about error handling any more. You just use the wrapper and you know that every MPI error is handled for you. This is a general concept, so if you use an MPI implementation that returns error codes, the wrapper will handle that for you.

Also, the wrapper has API for changing this behavior, i.e. returning error codes or terminating.